models/track_and_reid_model.py

```diff
@@ -0,0 +1,168 @@
+
+from fastreid.config import get_cfg
+from fastreid.data import build_reid_test_loader
+from demo.predictor import FeatureExtractionDemo
+from mmtrack.apis import inference_mot, init_model
+
+
+
+def get_args():
+    parser = ArgumentParser()
+    parser.add_argument('track_config', help='config file for the tracking model')
+    parser.add_argument('reid_config', help='config file for the reID model')
+    parser.add_argument('--input', help='input video file or folder')
+    parser.add_argument('--output', help='output video file (mp4 format) or folder')
+    parser.add_argument('--track_checkpoint', help='checkpoint file for the track model')
+    parser.add_argument('--reid_checkpoint', help='checkpoint file for the reID model')
+    parser.add_argument('--device', default='cuda:0', help='device used for inference')
+    parser.add_argument('--show', action='store_true', help='whether show the results on the fly')
+    parser.add_argument('--backend', choices=['cv2', 'plt'], default='cv2', help='the backend to visualize the results')
+    parser.add_argument('--fps', help='FPS of the output video')
+    parser.add_argument('--crops_folder', help='Path to the folder in which the generated crops should be saved')
+    parser.add_argument("--reid_opts", help="Modify reid-config options using the command-line 'KEY VALUE' pairs", default=[], nargs=REMAINDER,)
+    args = parser.parse_args()
+    return args
+
+
+def set_reid_cfgs(args):
+    cfg = get_cfg()
+    cfg.merge_from_file(args.reid_config)
+    cfg.merge_from_list(args.reid_opts)
+    cfg.freeze()
+    return cfg
+
+
+def apply_reid_model(reid_model, data):
+    feats = []
+    pids = []
+    camids = []
+    print('Converting test data to feature vectors:')
+    # Converts all images in the bounding_box_test and query to feature vectors
+    for (feat, pid, camid) in tqdm.tqdm(reid_model.run_on_loader(data), total=len(data)):
+        feats.append(feat)
+        pids.extend(pid)
+        camids.extend(camid)
+
+    print("The size of test gallery is", len(pids))
+    feats = torch.cat(feats, dim=0)
+    g_feat = feats
+    g_pids = np.asarray(pids)
+    # g_camids = np.asarray(camids)  # todo: remove?
+    return feats, g_feat, g_pids
+
+
+def find_best_match(q_feat, g_feat, g_pids):
+    """
+    Given feature vectors of the query images, return the ids of the images that are most similar in the test gallery
+    """
+    features = F.normalize(q_feat, p=2, dim=1)
+    others = F.normalize(g_feat, p=2, dim=1)
+    distmat = 1 - torch.mm(features, others.t())
+
+    distmat = distmat.numpy()
+    best_match_in_gallery = np.argmin(distmat, axis=1)
+    return g_pids[best_match_in_gallery]
+
+
+def tracking_inference(tracking_model, img, frame_id, acc_threshold=0.8):
+    result = inference_mot(tracking_model, img, frame_id=frame_id)
+    # result['track_bboxes'] = result.get('track_results')
+    acc = result['track_results'][0][:, -1]
+    mask = np.where(acc > acc_threshold)
```

Comment:
acc_threshold - do we send it as a hyper param for tunning in the original model_runner? if not add it